### PR TITLE
Fix error in copy constructor

### PR DESCRIPTION
**Pull request by ianagbip1oti**
The array copy appears to have a bug (i is never incremented). Using toArray instead.

roleIDs could probably be changed to a `List` or `Set`. It's Arrayness (real word, I swear) doesn't seem to be important.

### CODE REVIEW

**Comment by ianagbip1oti** [comment]
Just wait on this one. It's showing more changes than it should

**Comment by ianagbip1oti** [comment]
Ok, good now. Needed a merge

File: src/main/java/io/discloader/discloader/core/entity/guild/GuildMember.java [hunk 1]
```diff
@@ -108,11 +108,7 @@ public GuildMember(IGuildMember member) {
 		user = member.getUser();
 		guild = member.getGuild();
 		nick = member.getNickname();
-		roleIDs = new String[member.getRoles().size()];
-		int i = 0;
-		for (IRole role : member.getRoles().values()) {
-			roleIDs[i] = role.getID();
-		}
+		roleIDs = member.getRoles().values().toArray(new String[] { });
```
Comment (R3alCl0ud) [inline src/main/java/io/discloader/discloader/core/entity/guild/GuildMember.java]: this won't work. needs to be `member.getRoles().keySet().toArray(new String[] {});`

**Review by R3alCl0ud — CHANGES_REQUESTED** [review]

File: src/main/java/io/discloader/discloader/core/entity/guild/GuildMember.java (hunk 1 above)
Comment (R3alCl0ud) [reply src/main/java/io/discloader/discloader/core/entity/guild/GuildMember.java]: this code currently stringifies `IRole` objects

**Review by R3alCl0ud — APPROVED** [review]

File: src/main/java/io/discloader/discloader/core/entity/guild/GuildMember.java (hunk 1 above)
Comment (ianagbip1oti) [reply src/main/java/io/discloader/discloader/core/entity/guild/GuildMember.java]: Oops, just copied the values from the for loop. Fixed

File: src/main/java/io/discloader/discloader/core/entity/guild/GuildMember.java [hunk 2]
```diff
@@ -226,7 +231,9 @@ public IPermission getPermissions() {
 	 */
 	@Override
 	public IPresence getPresence() {
-		return guild.getPresences().get(getID());
+		if (presence == null)
+			presence = guild.getPresences().get(getID());
+		return presence;
```
Comment (R3alCl0ud) [inline src/main/java/io/discloader/discloader/core/entity/guild/GuildMember.java]: ??? this was added back because????

**Review by R3alCl0ud — CHANGES_REQUESTED** [review]

File: src/main/java/io/discloader/discloader/core/entity/guild/GuildMember.java (hunk 2 above)
Comment (ianagbip1oti) [reply src/main/java/io/discloader/discloader/core/entity/guild/GuildMember.java]: Not sure. Trying to work out why that's back

**Comment by ianagbip1oti** [comment]
Looks better now

**Comment by R3alCl0ud** [comment]
the revert in the `getPresence()` code is still there.

**Comment by ianagbip1oti** [comment]
Weird, I'm just seeing the role id change. Perhaps I'll just create a different PR

**Review by R3alCl0ud — APPROVED** [review]

**Pull request merged**